engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/BpmnElementProcessor.java

```diff
@@ -71,8 +86,11 @@ default void onActivate(final T element, final BpmnElementContext context) {}
    *
    * @param element the instance of the BPMN element that is executed
    * @param context process instance-related data of the element that is executed
+   * @return Either<Failure, Void> indicating the outcome of the completion attempt
    */
-  default void onComplete(final T element, final BpmnElementContext context) {}
+  default Either<Failure, Void> onComplete(final T element, final BpmnElementContext context) {
+    return EMPTY_RIGHT;
+  }
 
   /**
    * The element is going to be terminated. Perform every action to terminate the element and
```

Comment:
Regarding the `default void onTerminate(final T element, final BpmnElementContext context) {}` method, I noticed that we typically don't include incident creation logic there. 
@saig0 what do you think, would it be worth to also add an `Either<Failure, Void>` return type to it?